packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -619,33 +620,70 @@ class FlutterPlugin implements Plugin<Project> {
             addApiDependencies(project, variant.name, project.files {
                 packFlutterAppAotTask
             })
-            // We know that the flutter app is a subproject in another Android app when these tasks exist.
             Task packageAssets = project.tasks.findByPath(":flutter:package${variant.name.capitalize()}Assets")
             Task cleanPackageAssets = project.tasks.findByPath(":flutter:cleanPackage${variant.name.capitalize()}Assets")
+            // In add to app scenarios, :flutter is a subproject of another Android app.
+            // We know that :flutter is used as a subproject when these tasks exist.
+            boolean isUsedAsSubproject = packageAssets && cleanPackageAssets
             Task copyFlutterAssetsTask = project.tasks.create(name: "copyFlutterAssets${variant.name.capitalize()}", type: Copy) {
                 dependsOn compileTasks
-                if (packageAssets && cleanPackageAssets) {
+                compileTasks.each { flutterTask ->
+                    // Add flutter_assets.
+                    with flutterTask.assets
+                }
+                if (isUsedAsSubproject) {
                     dependsOn packageAssets
                     dependsOn cleanPackageAssets
                     into packageAssets.outputDir
-                } else {
-                    // `variant.mergeAssets` will be removed at the end of 2019.
-                    def mergeAssets = variant.hasProperty("mergeAssetsProvider") ?
-                        variant.mergeAssetsProvider.get() : variant.mergeAssets
-                    dependsOn mergeAssets
-                    dependsOn "clean${mergeAssets.name.capitalize()}"
-                    mergeAssets.mustRunAfter("clean${mergeAssets.name.capitalize()}")
-                    into mergeAssets.outputDir
+                    return
                 }
-                compileTasks.each { flutterTask ->
-                    with flutterTask.assets
+                // `variant.mergeAssets` will be removed at the end of 2019.
+                def mergeAssets = variant.hasProperty("mergeAssetsProvider") ?
+                    variant.mergeAssetsProvider.get() : variant.mergeAssets
+                dependsOn mergeAssets
+                dependsOn "clean${mergeAssets.name.capitalize()}"
+                mergeAssets.mustRunAfter("clean${mergeAssets.name.capitalize()}")
+                into mergeAssets.outputDir
+            }
+            if (!isUsedAsSubproject) {
+                variant.mergeResources.dependsOn(copyFlutterAssetsTask)
+                return
+            }
+            // Flutter module included as a subproject in add to app.
+            Project appProject = project.rootProject.findProject(':app')
+            assert appProject != null
+            appProject.afterEvaluate {
```

Comment:
I know enough to know that `afterEvaluate` is generally not recommended in Gradle scripts because it works outside of Gradle's normal execution flow, but I don't know enough about what to recommend here instead. :| If this is the only thing that works I think it makes sense to use it, but if there's another option I would prefer that instead.

Reply:
yeah. It needs to wait until the Android plugin loads. Otherwise `appProject.android` is undefined ☹️

Looking through [the docs](https://docs.gradle.org/current/userguide/build_lifecycle.html#sec:project_evaluation), it seems like `afterEvaluate` is the only way to perform additional configuration after all the definitions in a build script have been applied. I'm not sure if there's a better way other not adding the implicit dependency on AGP. However, that isn't an option here.